lib/lwm2m_carrier/os/lwm2m_carrier.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <lwm2m_carrier.h>
+
+#include "bootstrap_psk.h"
+
+BUILD_ASSERT_MSG((sizeof(CONFIG_LWM2M_CARRIER_BOOTSTRAP_URI) == 1) ||
+		 (sizeof(bootstrap_psk) > 0),
+		 "Custom bootsrap URI specified, please provide a valid PSK.");
+
+#define LWM2M_CARRIER_THREAD_STACK_SIZE 8192
+#define LWLM2_CARRIER_THREAD_PRIORITY K_LOWEST_APPLICATION_THREAD_PRIO
+
+__weak void lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
+{
+	ARG_UNUSED(event);
+}
+
+static lwm2m_carrier_config_t config;
+
+void lwm2m_carrier_thread_run(void)
+{
+	int err;
+
+	if (strlen(CONFIG_LWM2M_CARRIER_BOOTSTRAP_URI) > 0) {
+		config.bootstrap_uri = CONFIG_LWM2M_CARRIER_BOOTSTRAP_URI;
+	}
+
+	if (sizeof(bootstrap_psk) > 0) {
+		config.psk = (char *)bootstrap_psk;
+		config.psk_length = sizeof(bootstrap_psk);
+	}
+
+	err = lwm2m_carrier_init(&config);
+	__ASSERT(err == 0, "Failed to initialize LwM2M carrier library");
+
+	if (err != 0) {
+		return;
+	}
+
+	/* Non-return function. */
+	lwm2m_carrier_run();
```

Comment:
Drop the comment and put

	CODE_UNREACHABLE;

after lwm2m_carrier_run();

Then both the compiler and the reader will know that it is a "Non-return function".

Reply:
Right, done.